gossip/src/cluster_info.rs

```diff
@@ -817,7 +825,8 @@ impl ClusterInfo {
     /// If there exists a newer vote in gossip than `new_vote_slot` return `None` as this indicates
     /// that we might be submitting slashable votes after an improper restart
     fn find_vote_index_to_evict(&self, new_vote_slot: Slot) -> Option<u8> {
-        let self_pubkey = self.id();
+        let self_keypair = self.keypair();
+        let self_pubkey = self_keypair.pubkey();
```

Comment:
we don't need keypair here.
```suggestion
        let self_pubkey = self.id();
```